cohortextractor/tpp_backend.py

```diff
@@ -1335,7 +1335,9 @@ def patients_date_deregistered_from_all_supported_practices(self, between):
             max_date = "3000-01-01"
         if min_date is None:
             min_date = "1900-01-01"
-        date_condition, date_joins = self.get_date_condition("t", "t.end_date", between)
```

Comment:
Should a linter have picked up that `max_date` and `min_date` were never used?

Reply:
Sadly not, because they appear in conditionals. So this is a pyflakes error:
```py
def foo():
    bar = 1
```
But this isn't:
```py
def foo():
    bar = 1
    if bar:
        pass
```